src/server/pps/cmds/cmds.go

```diff
@@ -1815,11 +1814,39 @@ func pipelineHelper(ctx context.Context, pachctlCfg *pachctl.Config, reprocess b
 			request.Pipeline.Project = &pfs.Project{Name: project}
 		}
 
+		js, err := protojson.Marshal(request)
+		if err != nil {
+			return errors.Wrap(err, "could not convert request to JSON")
+		}
+		v2Req := &pps.CreatePipelineV2Request{
+			CreatePipelineRequestJson: string(js),
+			Update:                    request.Update,
+			Reprocess:                 request.Reprocess,
+			DryRun:                    request.DryRun,
+		}
+		if update {
+			v2Req.Update = true
+			v2Req.Reprocess = reprocess
+		}
+		v2Req.DryRun = dryRun
 		if err = txncmds.WithActiveTransaction(pc, func(txClient *pachdclient.APIClient) error {
-			_, err := txClient.PpsAPIClient.CreatePipeline(
+			resp, err := txClient.PpsAPIClient.CreatePipelineV2(
 				txClient.Ctx(),
-				request,
+				v2Req,
 			)
+			if dryRun {
+				d := json.NewDecoder(strings.NewReader(resp.EffectiveCreatePipelineRequestJson))
+				d.UseNumber()
+				var effectiveSpec any
+				if err := d.Decode(&effectiveSpec); err != nil {
+					return errors.Wrapf(err, "could not decode effective spec %q", resp.EffectiveCreatePipelineRequestJson)
+				}
+				b, err := json.MarshalIndent(effectiveSpec, "", "\t")
```

Comment:
This probably wants an -o flag for consistency with "inspect pipeline".

Reply:
You’re right. I’ve added `-o`/`--output` as well as `--raw`, and pretty-printed output based on `pretty.PrintDetailedPipelineInfo`.